### ISSUE TITLE
Colorspace Improvement

### ISSUE THREAD

**Issue by ANekhai** [ConorLazarou/PokeGAN#10]
Hello! I want to preface by saying I found this work very inspirational. I really like the idea of transferring the style of real Pokemon to a latent variable. I have plans to explore some extensions to your technique soon with a larger dataset of roughly 5k Pokemon images. 

I wanted to highlight an improvement to your colorspace idea. I think what you have implemented is a manual method of doing matrix multiplication. I have implemented a method that only depends on PyTorch's matmul function. With a transposition and a colorspace tensor of the form (b, 1, 16, 3) you can directly do `intermediate_image @ color_tensor` to do all three color transformation blocks in a single operation! Note: this requires your intermediate image tensor to be of the form (b, im_size, im_size, 16) so you transpose the image before and after the operation. The seventh cell of [this notebook ](https://github.com/ANekhai/pokefun/blob/main/notebooks/colorspace_scratch.ipynb) highlights the math. 

Cheers and thank you for  your detailed writeups both on Medium and Arxiv!

**Comment by RileyLazarou** [comment]
You're right, thanks for the tip, that's much more efficient! Good luck with your exploration, I'd love to see any exciting findings you have!

**Comment by ANekhai** [comment]
I'll keep you posted!
